src/NimBLECharacteristic.cpp

```diff
@@ -459,6 +467,11 @@ void NimBLECharacteristic::notify(const uint8_t* value, size_t length, bool is_n
     int rc = 0;
 
     for (auto &it : m_subscribedVec) {
+        // check if need a specific client
+        if ((conn_handle >= 0) && (it.first != conn_handle)) {
```

Comment:
We should check for `(conn_handle > BLE_HCI_LE_CONN_HANDLE_MAX)` instead of `(conn_handle >= 0)` here. Please see the comment below about the default value for details.

Reply:
corrected as per comment